### ISSUE TITLE
Staking Wallet Minting Bug

### ISSUE THREAD

**Issue by doncrypto** [monyxcoin/monyx-coin#111]
From observations it appears your minting rate is off. Can you check this so we can receive the accurate rate at 10%/month? 0.33%/day

windows wallet 3.0.0.1

Thanks,
DC

**Comment by Sirbearingson** [comment]
@monyxcoin

**Comment by monyxcoin** [comment]
@doncrypto sure, we'll check it

**Comment by monyxcoin** [comment]
@doncrypto PoS reward scales with the coin age, it's made like so because 1) otherwise the total 24m limit of coins will be depleted really fast, 2) with more and more wallets running there will be more and more possible candidates for a new PoS block, so all other possible candidates getting older and will receive bigger reward (scaled with the coin age).
That's how it's done in every coin, as for example davorcoin: https://github.com/davorcc/DavorCoin/blob/master/src/main.cpp#L1012.

You can see the algorithm here:
https://github.com/monyxcoin/monyx-coin/blob/master/src/main.cpp#L973

**Comment by doncrypto** [comment]
The 10%/month number on whitepaper will be rewarded to the most coinage and not everyone that stakes will get the 10%/month?

Also another staking question,
I recently got my first stake and then my coinage was reset, I have to mature my coins again(purple), and then my  staking amounts were split into 2 inputs when I had combined them previously. Is this normal behavior? Is there a limit an input should stake?

Thanks for the reply.
